testcases/open_posix_testsuite/conformance/interfaces/feof/1-1.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * This file is licensed under the GPL license.  For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ *
+ *	The feof function shall test the end-of-file indicator for
+ *	the stream pointed to by argument and the function shall return
+ *	non-zero if and only if the end-of-file indicator is set for stream
+ *
+ * method:
+ *	-open file in write mode
+ *	-write 2 sample strings into the file
+ *	-close the file
+ *	-open the same file in read mode
+ *	-read first string from file and check feof return value
+ *	-read second string from file and check feof return value
+ */
+
+#define _XOPEN_SOURCE 700
+#include <stdio.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include "posixtest.h"
+
+#define TNAME "feof/1-1.c"
+#define FNAME "in_file"
+#define STRING_CONST1 "LINUX"
+#define STRING_CONST2 "POSIX"
+
+static void remove_file(FILE *in_fp)
+{
+	errno = 0;
+	if (in_fp != NULL) {
+		if (fclose(in_fp) == EOF) {
+			printf(TNAME " Error at fclose(), errno = %d\n", errno);
+			unlink(FNAME);
+			exit(PTS_UNRESOLVED);
+		}
+	}
+	unlink(FNAME);
+}
+
+int main(void)
+{
+	FILE *in_fp;
+	char sample_string1[strlen(STRING_CONST1)];
+	char sample_string2[strlen(STRING_CONST2)];
```

Comment:
This is wrong, the array is not large enough to hold the string, you should use sizeof(STRING) which will include space for the terminating zero byte.

Also C is a spartan language and so the identifiers should be short and to the point, there is no need to include the CONST part in the STRING_CONST1 and STRING is commonly shortened to just STR, so these may as well be STR1 and STR2 and similarily for other identifiers...

Reply:
> This is wrong, the array is not large enough to hold the string, you should use sizeof(STRING) which will include space for the terminating zero byte.

Totally agree!

> Also C is a spartan language and so the identifiers should be short and to the point, there is no need to include the CONST part in the STRING_CONST1 and STRING is commonly shortened to just STR, so these may as well be STR1 and STR2 and similarily for other identifiers...

I disagree on this point: having well-named variables makes them self-documenting, so you don't need to provide comments and someone reading your code can better understand the flow and intent for the code. I can't tell you how many times I (and others) have had to refactor tests because we didn't understand the intent of the original implementation (no comments and poorly named variables and functions).

Reply:
Updated to sizeof(STRING) for the length of array.
 >Also C is a spartan language and so the identifiers should be short and to the point, there is no need to include the CONST part in the STRING_CONST1 and STRING is commonly shortened to just STR, so these may as well be STR1 and STR2 and similarily for other identifiers...
i have made it little simpler as STR_CONST#